### ISSUE TITLE
Missing backing image after performing resize on unshelved instances.

### ISSUE THREAD

**Issue by pmesserli1** [rcbops/rpc-openstack#1115]
##### What version (commit) do you encounter this issue on?
/opt/rpc-openstack/rpcd# git describe --tags
r12.0.1-1-gec81ea1
##### What type of issue is this (bug, feature, documentation, regression)?
Bug
##### What is the expected behavior?
After instances are unshelved, I should be able to resize without error.

##### What is actually happening?
If I create a new instance and then shelve it, I believe the _base image gets removed from the compute node.  After I unshelve this instance and initiated a resize the instance goes into error with the qemu-img resize returning exit code 1.  If you run this on the compute it fails due to the UUID image not being in _base.  

qemu-img resize /var/lib/nova/instances/81020327-04fc-4810-87d1-24fb9fc09cf1/disk 42949672960
qemu-img: Could not open '/var/lib/nova/instances/81020327-04fc-4810-87d1-24fb9fc09cf1/disk': Could not open backing file: Could not open '/var/lib/nova/instances/_base/2a3474b75664224eead69c257b08c4d9f12e3c26': No such file or directory

If you touch the file in the dir, it would complete since the resize just looks to see if this file exists not actually use it.

##### How can we reproduce this issue?
Create new instance
Nova shelve instance
Nova unshelve instance
Nova resize instance

##### How can we verify your issue is addressed?
Resize completes after a instance is unshelved

##### What is your evaluation on the importance of this issue, and why?
If users are shelving instances and intend to resize they will run into errors

###### Remove branches (lines) you do not wish to target.
- [ ] Master
- [ ] liberty-12.2
- [ ] liberty-12.1
- [ ] kilo

**Comment by git-harry** [comment]
r12.0.1-1-gec81ea1 indicates there is an additional commit on top of r12.0.1 - what is it?

After the instance is unshelved but before you resize it does it work? Are you able to ssh to it?

To clarify, is the result of attempting the resize data loss?

**Comment by d34dh0r53** [comment]
This will probably need to be resolved upstream, and we'll need more information about this.  We can re-prioritize later if it is deemed that this is impacting our customers.

**Comment by pmesserli1** [comment]
After the instance is unshelved everything is ok and you are able to SSH to the instance.  However if you initiate a resize it fails due to the _base file for the instance missing, causing the instance to go into a error state.  It does not actually cause any data loss or prevent you from ssh'ing to the instance.  You would have to reset the state of the instance to active, and then touch a file under _base with the instance UUID and try the resize again.

**Comment by feli5** [comment]
I also encountered this problem, I'm using redhat OSP9
# rpm -qa |grep nova
openstack-nova-cert-13.1.1-2.el7ost.noarch
openstack-nova-api-13.1.1-2.el7ost.noarch
openstack-nova-console-13.1.1-2.el7ost.noarch
openstack-nova-common-13.1.1-2.el7ost.noarch
openstack-nova-scheduler-13.1.1-2.el7ost.noarch
python-novaclient-3.3.1-1.el7ost.noarch
python-nova-13.1.1-2.el7ost.noarch
openstack-nova-conductor-13.1.1-2.el7ost.noarch
openstack-nova-novncproxy-13.1.1-2.el7ost.noarch

**Comment by byronmccollum** [comment]
When you shelve an instance, is the shelved "image" flattened with the _base image? Seems like unshelving __*should*__ place the shelved "image" in _base, and act as the new backing image for the respawned instance...but actually doesn't.
